tests/lib/rules/prop-types.js

```diff
@@ -3533,6 +3524,18 @@ ruleTester.run('prop-types', rule, {
         message: '\'fooBar\' is missing in props validation'
       }],
       parser: 'babel-eslint'
+    }, {
+      code: [
+        'function Hello({names}) {',
+        '  return names.map((name) => {',
+        '    return <div>{name}</div>;',
+        '  });',
+        '}'
+      ].join('\n'),
+      errors: [{
+        message: '\'names\' is missing in props validation'
```

Comment:
this is a good fix; but this should be able to go in a separate PR

Reply:
It's a by-product of checking `map`. Considering this request and your concern above, should I remove the `map` support for now with it being added later as an enhancement?

Reply:
In the case of the `prop-types` rule, `names` isn't inside a `map`, so it should be caught regardless.

Reply:
It doesn't have to do with the JSX being returned in `map`. The reason it's being caught now is because `isReturningJSX` wasn't considering functions with return statements that are calling `Array#map`. So, in order to have this test case have an error, it required the same change that was made to check for returning `Array#map` in a component class method.

Reply:
Right - I'm saying that the bug fix for `prop-types` can be to consider these return statements, even if it ignores the code inside the `.map` callback.

Reply:
Okay, so in order to still support the `Array#map` support for this rule, you want me to find a solution that makes it work while also not resolving the issue with that particular test case?

Reply:
I'm thinking that the `prop-types` issue can be solved without having to guess that `.map` is "array map".

Then separately, this rule may be able to avoid making the same guess.